src/components/common/TermsBanner/index.tsx

```diff
@@ -0,0 +1,61 @@
+import { AppRoutes } from '@/config/routes'
+import css from './styles.module.css'
+import { Button, Typography } from '@mui/material'
+import Link from 'next/link'
+import MUILink from '@mui/material/Link'
+import useLocalStorage from '@/services/local-storage/useLocalStorage'
+import { useEffect, useMemo, useState } from 'react'
+import { type CookiesState, CookieType } from '@/store/cookiesSlice'
+import local from '@/services/local-storage/local'
+
+const TERMS_KEY = 'terms_dismissed'
+
+const isExistingUser = () => {
+  return local.getItem<CookiesState>('cookies')?.[CookieType.NECESSARY] || false
+}
+
+const TermsBanner = () => {
+  const [isMounted, setIsMounted] = useState<boolean>(false)
+  const [isDismissed = false, setIsDismissed] = useLocalStorage<boolean>(TERMS_KEY)
+
+  // Check on page load if "necessary" cookies have been accepted
+  const existingUser = useMemo(() => isExistingUser(), [])
+  const shouldOpen = existingUser && !isDismissed
+
+  useEffect(() => {
+    setIsMounted(true)
+  }, [])
```

Comment:
Can we not instantiate the state `true`?

Reply:
It didn't work. We need to wait until the component is mounted before we set the value.

Reply:
I removed the state and import `TermsBanner` via `next/dynamic` with ssr turned off for this component to solve the hydration error.